src/jit/importer.cpp

```diff
@@ -2207,6 +2207,13 @@ bool Compiler::impSpillStackEntry(unsigned level,
     /* Assign the spilled entry to the temp */
     impAssignTempGen(tnum, tree, verCurrentState.esStack[level].seTypeInfo.GetClassHandle(), level);
 
+    // If temp is single def and a ref type, grab what type info we can.
+    if (lvaTable[tnum].lvIsTemp && (lvaTable[tnum].lvType == TYP_REF))
+    {
+        CORINFO_CLASS_HANDLE stkHnd = verCurrentState.esStack[level].seTypeInfo.GetClassHandle();
+        lvaSetClass(tnum, tree, stkHnd);
+    }
```

Comment:
I did see a comment in the code base that lvIsTemp variable are single def, but there is nothing that enforces that and I spotted at least in one place where that isn't true.

` genReturnLocal = lvaGrabTemp(true DEBUGARG("Single return block return value"));``